pkg/planner/core/operator/logicalop/logical_aggregation.go

```diff
@@ -632,44 +632,96 @@ func (la *LogicalAggregation) splitCondForAggregation(predicates []expression.Ex
 		exprsOriginal = append(exprsOriginal, fun.Args[0])
 	}
 	groupByColumns := expression.NewSchema(la.GetGroupByCols()...)
+	aggColumns := expression.NewSchema(la.getAggFuncsColsForFirstRow()...)
 	// It's almost the same as pushDownCNFPredicatesForAggregation, except that the condition is a slice.
 	for _, cond := range predicates {
-		subCondsToPush, subRet := la.pushDownDNFPredicatesForAggregation(cond, groupByColumns, exprsOriginal)
+		subCondsToPush, subRet := la.pushDownDNFPredicates(cond, groupByColumns, exprsOriginal, la.pushDownPredicatesByGroupby)
 		if len(subCondsToPush) > 0 {
 			condsToPush = append(condsToPush, subCondsToPush...)
 		}
 		if len(subRet) > 0 {
-			ret = append(ret, subRet...)
+			for _, s := range subRet {
+				subCondsToPush1, subRet1 := la.pushDownDNFPredicates(s, aggColumns, exprsOriginal, la.pushDownPredicatesByAggFuncs)
```

Comment:
Please add more comments about this part, for example, why we have to check `subRet` again, and which problems this code can solve.

Reply:
I have add the comment.